logstash-core/src/main/java/org/logstash/instrument/witness/PluginsWitness.java

```diff
@@ -69,5 +94,59 @@ private PluginWitness getPlugin(Map<String, PluginWitness> plugin, String id) {
         return plugin.computeIfAbsent(id, k -> new PluginWitness(k) );
     }
 
-}
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        new Serializer().innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    public static class Serializer extends StdSerializer<PluginsWitness> {
+
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(PluginsWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<PluginsWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(PluginsWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
 
+        void innerSerialize(PluginsWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeObjectFieldStart(KEY);
+
+            serializePlugins("inputs", witness.inputs, gen, provider);
+            serializePlugins("filters", witness.filters, gen, provider);
+            serializePlugins("outputs", witness.outputs, gen, provider);
+            //codec is not serialized
+
+            gen.writeEndObject();
+        }
+
+        private void serializePlugins(String key, Map<String, PluginWitness> plugin, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeArrayFieldStart(key);
+            for (Map.Entry<String, PluginWitness> entry : plugin.entrySet()) {
```

Comment:
@jakelandis shouldn't this just work out of the box if we have a `Serializer` for `PluginWitness ` registered with Jackson? Meaning, we should simply be able to `com.fasterxml.jackson.core.JsonGenerator#writeObject` the `plugin` variable? (this comment applies to all the similar spots we have in the code as well)